app/controllers/application_controller.rb

```diff
@@ -179,17 +179,33 @@ def on_main_domain?
     request.host == Figaro.env.domain || request.host == "app.#{Figaro.env.domain}"
   end
 
-  def current_delivery
-    return nil if (current_market.blank? || current_organization.blank?) && session[:order_id].nil?
+  def require_current_delivery
+    if current_delivery.blank?
+      # falls through to redirect
 
-    if session[:order_id] # We're adding an item to an order, so use the delivery of the order
-      @current_delivery = Order.find(session[:order_id]).delivery.decorate
+    # Ask Rob if we want to preserve ability to order without pickup OR delivery!!!
+    # elsif current_delivery.requires_location? && selected_organization_location.nil?
+    #   return
+    elsif !current_delivery.can_accept_orders?
```

Comment:
@thermistor if it is in the existing code then likely we want to retain. Looking at `Delivery::requires_location?` it seems to say "if this is not a buyer pickup" then we require a location. In other words if this is a delivery then we need the buyer to have set an address for us to deliver to. So I think yes we want to retain that and should have specs covering that behavior.

Reply:
This line translates roughly as "if not a buyer pickup, then you need a location, except when you don't have a location". It's a really old commit (2014-ish). The requires_location? is used elsewhere, so this might be just a localized exception to the general business rule, but if uncommented would allow people to purchase without a pickup or delivery. Can you think of circumstances where that makes sense?